src/main/scala/viper/silver/parser/ParseAst.scala

```diff
@@ -264,23 +266,24 @@ object PTypeVar{
   val sep = "#"
   //TODO: do this properly
   def isFreePTVName(s : String) = s.contains(sep)
-  private var lastIndex = 0
+  private val lastIndex = new AtomicInteger(0)
   //Generate a unique fresh version of old
   def fresh(old: PDomainType) = {
     require(old.isTypeVar)
-    val freshName = getFreshName(old.domain.name)
-    lastIndex+=1
+    val ind = lastIndex.getAndIncrement()
+    val freshName = getFreshName(old.domain.name, ind)
     PTypeVar(freshName)
   }
```

Comment:
My IDE is reporting that this method (`fresh`) does not seem to be used. (Same for `freshTypeSubstitutionPTVs` which also contains the only use of `freshTypeSubstitution`). It may be that my IDE is set-up in a weird way, but if these are indeed the only cases, couldn't we just drop this code instead?

Reply:
There is a usage of ``freshTypeSubstitutionPTVs`` in the ADT plugin, and ``freshTypeSubstitution`` is used in the Resolver and in the ADT plugin. I also don't see any usage of ``fresh`` anywhere, but I guess there's a chance it's used e.g. in some other plugin that's not inside the Silver repo, so I'd rather keep it around just in case.